### ISSUE TITLE
Badips.py incompatible with IPv6 (Breaks ban actions with actionstart_on_demand).

### ISSUE THREAD

**Issue by PatTheMav** [fail2ban/fail2ban#2390]
### Environment:

- Fail2Ban 0.10.2-2.1
- Debian GNU/Linux 9 (stretch)
- [x] Fail2Ban installed via OS/distribution mechanisms
- [X] You have not applied any additional foreign patches to the codebase
- [X] Some customizations were done to the configuration (provide details below is so)

### The issue:

`Badips.py` does not specify a `family` when issuing bans, thus `actionstart_on_demand` is not triggered for ban actions.

As bans are restored on startup (and before fail2ban restores its own bans), this will fail for every IP received from badips.com as the necessary filters and sets are not created e.g. for `nftables`.

After 900 seconds fail2ban will retry restoring the bans from badips.com. If a normal ban has happened (or old bans have been restored), this will succeed.

If no regular bans happened, all errors will be repeated.

#### Steps to reproduce

* Set banaction to `nftables-multiport`
* Add `badips.py[category="ssh", banaction="%(banaction)s", agent="%(fail2ban_agent)s"]` to actions in jail `sshd`.
* (Re)start fail2ban

#### Expected behavior

* badips.py will check for `actionstart_on_demand` and execute the `actionstart` from given `banaction` if necessary
* Bans from badips.com are established without requiring a `banaction` from another action.

#### Observed behavior

See explanation under "The Issue".

#### Relevant parts of /var/log/fail2ban.log file:

```
2019-03-31 20:14:21,056 fail2ban.BadIPsAction   [15040]: ERROR   Error banning IP 125.74.10.146 for jail 'sshd' with action 'nftables-multiport': Can't conve
rt 'NoneType' object to str implicitly
Traceback (most recent call last):
  File "/etc/fail2ban/action.d/badips.py", line 279, in _banIPs
    'ipjailmatches': "",
  File "/usr/lib/python3/dist-packages/fail2ban/server/action.py", line 434, in ban
    cmd = self._getOperation('<actionstart>', family)
  File "/usr/lib/python3/dist-packages/fail2ban/server/action.py", line 358, in _getOperation
    conditional=('family=' + family), cache=self.__substCache)
TypeError: Can't convert 'NoneType' object to str implicitly
```

As no `family` is specified inside `badips.py`, the `actionstart` is not executed and the banaction is not properly initialized:

```
2019-03-31 19:21:10,131 fail2ban.action         [29966]: DEBUG   nft list chain inet fail2ban INPUT | grep -q '@f2b-sshd[ \t]'
2019-03-31 19:21:10,169 fail2ban.utils          [29966]: Level 39 7f6a582dc7a0 -- exec: nft list chain inet fail2ban INPUT | grep -q '@f2b-sshd[ \t]'
2019-03-31 19:21:10,170 fail2ban.utils          [29966]: ERROR   7f6a582dc7a0 -- returned 1
2019-03-31 19:21:10,170 fail2ban.CommandAction  [29966]: ERROR   Invariant check failed. Trying to restore a sane environment
```

**Comment by PatTheMav** [comment]
I "fixed" this issue by adding `inet4` as `family` to `_banIPs` and `_unbanIPs` and filtering out any IPv6 addresses that might be received from badips.com (using `ips = {ip for ip in ips if ':' not in ip}` inside the `update` function).

This is a band-aid until a proper fix has been found.

**Comment by sebres** [comment]
Should be solved now in 22b93045629f207a5352be8abfc099024f0eab0d.
I'll merge it to major branches if the tests pass.
Thanks!
